File: src/main/java/com/twb/designpatternsplugin/listeners/MyProjectManagerListener.java
package com.twb.designpatternsplugin.listeners;

import com.intellij.openapi.project.Project;
import com.intellij.openapi.project.ProjectManagerListener;
import org.apache.commons.lang3.builder.ReflectionToStringBuilder;
import org.jetbrains.annotations.NotNull;

public class MyProjectManagerListener implements ProjectManagerListener {

    @Override
    public void projectOpened(@NotNull Project project) {
        System.out.println("project = " + ReflectionToStringBuilder.toString(project));

    }
}

File: src/main/kotlin/com/github/thomasbigger584/designpatternsplugin/listeners/MyProjectManagerListener.kt
package com.github.thomasbigger584.designpatternsplugin.listeners

import com.intellij.openapi.components.service
import com.intellij.openapi.project.Project
import com.intellij.openapi.project.ProjectManagerListener
import com.github.thomasbigger584.designpatternsplugin.services.MyProjectService

internal class MyProjectManagerListener : ProjectManagerListener {

    override fun projectOpened(project: Project) {
        project.service<MyProjectService>()
    }
}
